packages/components/tree/src/flat-tree-data-source.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  TreeDataSource,
+  type TreeDataSourceMapping,
+  TreeDataSourceNode,
+  type TreeDataSourceOptions
+} from './tree-data-source.js';
+
+export interface FlatTreeDataSourceMapping<T> extends TreeDataSourceMapping<T> {
+  getLevel(item: T): number;
+}
+
+export interface FlatTreeDataSourceOptions<T> extends FlatTreeDataSourceMapping<T> {
+  loadChildren?(node: T): Promise<T[]>;
+  selects?: 'single' | 'multiple';
+}
+
+/**
+ * A tree model that represents a flat list of nodes.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export class FlatTreeDataSource<T = any> extends TreeDataSource<T> {
+  #mapping: FlatTreeDataSourceMapping<T>;
```

Comment:
Can you please add spacing between variables and maybe add descriptions as well? :)